tools/clang/test/HLSL/ShaderOpArith.xml

```diff
@@ -3616,29 +3642,39 @@ void MSMain(uint GID : SV_GroupIndex,
     #else // NO FALLBACK
 
     #define ROOT_SIG [RootSignature("RootFlags(CBV_SRV_UAV_HEAP_DIRECTLY_INDEXED | SAMPLER_HEAP_DIRECTLY_INDEXED | ALLOW_INPUT_ASSEMBLER_INPUT_LAYOUT)")]
+    
+    static ConstantBuffer<Constants> CB = ResourceDescriptorHeap[12];
 
-    static ConstantBuffer<Constants> CB = ResourceDescriptorHeap[6];
-
-    static ByteAddressBuffer g_rawBuf = ResourceDescriptorHeap[CB.idx0];
-    static StructuredBuffer<float> g_structBuf = ResourceDescriptorHeap[CB.idx1];
-    static Texture2D<float> g_tex = ResourceDescriptorHeap[CB.idx2];
-    static RWByteAddressBuffer g_rwRawBuf = ResourceDescriptorHeap[CB.idx3];
-    static RWStructuredBuffer<float> g_rwStructBuf = ResourceDescriptorHeap[CB.idx4];
-    static RWTexture1D<float> g_rwTex = ResourceDescriptorHeap[CB.idx5];
+    #define NON_UNIFORM
```

Comment:
Option should be driven by a compilation argument I suspect, then code in test added to try both cases.